test/rekt/resources/apiserversource/apiserversource.go

```diff
@@ -166,3 +171,46 @@ func labelSelectorToStringMap(selector *metav1.LabelSelector) map[string]interfa
 
 	return r
 }
+
+func MatchLabels(labels1 map[string]string, labels2 map[string]string) bool {
+	for k, v := range labels1 {
+		if labels2[k] != v {
+			return false
+		}
+	}
+	return true
+}
+
+func TestLabels() map[string]string {
+	return map[string]string{
+		"testkey":  "testvalue",
+		"testkey1": "testvalue1",
+		"testkey2": "testvalue2",
+	}
+}
+
+func VerifyNodeSelectorDeployment(source string) feature.StepFn {
+	return func(ctx context.Context, t feature.T) {
+		env := environment.FromContext(ctx)
+		ns := env.Namespace()
+
+		kubeClient := client.Get(ctx)
+
+		deps, err := kubeClient.AppsV1().Deployments(ns).List(ctx, metav1.ListOptions{})
+		if err != nil {
+			t.Fatalf("error getting deployment: %v", err)
+		}
+
+		var dep appsv1.Deployment
+
+		for _, d := range deps.Items {
+			if kmeta.ChildName(fmt.Sprintf("apiserversource-%s-", source), string(d.GetUID())) == d.Name {
+				dep = d
+			}
+		}
```

Comment:
Is there a reason why you are listing and then looping over the list instead of just directly getting the deployment you want?

Reply:
ya because the name is calculated before its assigned for the deployment so I think we don't know with which name we have our deployment so I just tried to match the calculations and in that way we might get the right deployment